src/rtl/hbsocket.c

```diff
@@ -2848,6 +2860,21 @@ int hb_socketSetNoDelay( HB_SOCKET sd, HB_BOOL fNoDelay )
    return ret;
 }
 
+int hb_socketSetNoSigPipe( HB_SOCKET sd, HB_BOOL fNoSigPipe )
+{
+#if defined( SO_NOSIGPIPE )
+   int val = fNoSigPipe ? 1 : 0, ret;
+   ret = setsockopt( sd, SOL_SOCKET, SO_NOSIGPIPE, ( const char * ) &val, sizeof( val ) );
+   hb_socketSetOsError( ret != -1 ? 0 : HB_SOCK_GETERROR() );
```

Comment:
I'd parenthsize the condition to ease human suffering

Reply:
I have followed the other procedure to be the same way.